File: dcp-solutions/2024-09-18/solution.rb
### Ruby Solution
```markdown
def largest_sum_of_non_adjacent(nums)
  if nums.empty?
    return 0
  elsif nums.size <= 2
    return nums.max
  end

  # Initialize dynamic programming array with zeros
  dp = Array.new(nums.size, 0)

  # Base case: pick the maximum number in the first two numbers
  dp[0] = nums[0]
  dp[1] = [nums[0], nums[1]].max

  # Fill up the rest of the dynamic programming array
  for i in 2..-1 do
    dp[i] = [dp[i-1], dp[i-2] + nums[i]].max
  end

  # Return the maximum sum found
  dp[-1]
end
```

File: dcp-solutions/2024-09-18/solution.py
```python
def largest_sum_of_non_adjacent(numbers):
    if not numbers:
        return 0
    elif len(numbers) <= 2:
        return max(0, max(numbers))

    incl = [0] * len(numbers)
    excl = [0] * len(numbers)

    incl[0] = max(0, numbers[0])
    excl[0] = 0

    for i in range(1, len(numbers)):
        incl[i] = max(incl[i-1], excl[i-1] + numbers[i])
        excl[i] = max(excl[i-1], incl[i-1])

    return max(incl[-1], excl[-1])
```